oddt/fingerprints.py

```diff
@@ -405,15 +409,129 @@ def ECFP(mol, depth=2, size=4096, count_bits=True, sparse=True,
+                Column[0] = index of hashed atoms
+                Column[1] = neigbor's coords of ligand's atom,
+                (5, 3) -> ligand's atom coords and 4 his neigbors
+                Column[2] = neigbor's coords of protein's atom,
+                (5, 3) -> protein's atom coords and 4 his neigbors
+
+    """
+    # removing h
+    protein_dict = protein.atom_dict[protein.atom_dict['atomicnum'] != 1]
+    ligand_dict = ligand.atom_dict[ligand.atom_dict['atomicnum'] != 1]
+
+    protein_atoms, ligand_atoms = close_contacts(
+        protein_dict, ligand_dict, cutoff=4.5)
+    SPLIF = np.zeros((len(ligand_atoms)),
+                     dtype=[('hash', int), ('ligand_coords', np.float32, (5, 3)),
+                            ('protein_coords', np.float32, (5, 3))])
+    for index in range(len(ligand_atoms)):
+        SPLIF[index] = (hash32((
+            _ECFP_atom_hash(ligand, int(ligand_atoms[index]['id'] - 1), depth=depth)[-1],
+            _ECFP_atom_hash(protein, int(protein_atoms[index]['id'] - 1), depth=depth)[-1])),
+            np.vstack((ligand_atoms[index]['coords'].reshape((1, 3)),
+                       ligand_atoms[index]['neighbors'])),
+            np.vstack((protein_atoms[index]['coords'].reshape((1, 3)),
+                       protein_atoms[index]['neighbors'])))
+
+    # folding
+    SPLIF['hash'] = np.floor((np.array(SPLIF['hash']).astype(np.float64) - MIN_HASH_VALUE) /
+                             int(abs(MAX_HASH_VALUE - MIN_HASH_VALUE) / size))
+    return np.sort(SPLIF)
+
+
+def similarity_SPLIF(reference, query, rmsd_cutoff=1.):
+    """
+        Calculates similarity between structural interaction fingerprints,
+        based on doi:http://pubs.acs.org/doi/abs/10.1021/ci500319f.
+
+        Parameters
+        ----------
+        reference, query: numpy.array
+            SPLIFs, which are compared in order to determine similarity.
+        rmsd_cutoff : int (default = 1)
+            Specific treshold for which, bits are considered as fully matching.
+
+        Returns
+        -------
+        SimilarityScore : float
+            Similarity between given fingerprints.
+
+    """
+    result = []
+    # intersection of reference and query hashed atoms
+    index = np.intersect1d(reference['hash'], query['hash'])
+
+    ref_intersection = reference[np.where(np.in1d(reference['hash'], index))]
+    ref_group_intersection = np.split(ref_intersection, np.searchsorted(
+        ref_intersection['hash'], index[1:]))  # reference
+
+    query_intersection = query[np.where(np.in1d(query['hash'], index))]
+    query_group_intersection = np.split(query_intersection, np.searchsorted(
+        query_intersection['hash'], index[1:]))  # query
+
+    numla = 0  # number of unique matching ligand atoms
+    nula = 0  # number of unique ligand atoms
+    numpa = 0  # number of unique matching protein atoms
+    nupa = 0  # number of unique protein atoms
+
+    def rmsd(reference, query):
```

Comment:
I think it would be nice to have general-purpose RMSD somewhere outside this function (`oddt.spatial` maybe?).